src/gluonts/model/seq2seq/_forking_estimator.py

```diff
@@ -108,47 +115,107 @@ def __init__(
         self.quantile_output = quantile_output
         self.prediction_length = prediction_length
         self.freq = freq
+        self.use_dynamic_feat = use_dynamic_feat
         self.context_length = (
             context_length if context_length is not None else prediction_length
         )
+        self.add_time_feature = add_time_feature
+        self.add_age_feature = add_age_feature
+
+        # TODO: refactor this variable name: dynamic_network, in fact it
+        #  is not even necessary as is, because this is how use_dynamic_feat was
+        #  set in MQCNNEstimator and otherwise its not used, i.e. False
+        # is target only network or not?
+        self.dynamic_network = (
+            use_dynamic_feat or add_time_feature or add_age_feature
+        )
+        print(f"use_dynamic_network: {self.dynamic_network}")
 
     def create_transformation(self) -> Transformation:
-        return Chain(
-            trans=[
-                AsNumpyArray(field=FieldName.TARGET, expected_ndim=1),
-                ForkingSequenceSplitter(
-                    train_sampler=TestSplitSampler(),
-                    enc_len=self.context_length,
-                    dec_len=self.prediction_length,
+        chain = []
+        dynamic_feat_fields = []
+
+        if self.add_time_feature:
+            chain.append(
+                AddTimeFeatures(
+                    start_field=FieldName.START,
+                    target_field=FieldName.TARGET,
+                    output_field=FieldName.FEAT_TIME,
+                    time_features=time_features_from_frequency_str(self.freq),
+                    pred_length=self.prediction_length,
+                ),
+            )
+            dynamic_feat_fields.append(FieldName.FEAT_TIME)
+
+        if self.add_age_feature:
+            chain.append(
+                AddAgeFeature(
+                    target_field=FieldName.TARGET,
+                    output_field=FieldName.FEAT_AGE,
+                    pred_length=self.prediction_length,
                 ),
-            ]
+            )
+            dynamic_feat_fields.append(FieldName.FEAT_AGE)
+
+        if self.use_dynamic_feat:
+            dynamic_feat_fields.append(FieldName.FEAT_DYNAMIC_REAL)
+
+        if len(dynamic_feat_fields) > 1:
+            chain.append(
+                VstackFeatures(
+                    output_field=FieldName.FEAT_DYNAMIC_REAL,
+                    input_fields=dynamic_feat_fields,
+                )
+            )
+        elif len(dynamic_feat_fields) == 1:
+            chain.append(
+                RenameFields(
+                    {dynamic_feat_fields[0]: FieldName.FEAT_DYNAMIC_REAL}
+                )  # TODO: find out why this is done
+            )
```

Comment:
What is the purpose of this.?

Reply:
`FEAT_DYNAMIC_REAL` can consist of different features. If there is only one (for example `feat_age`, we rename the input field to be `feat_dynamic_real`.

Reply:
I see, thx!